templates/tftmpl/integration_test.go

```diff
@@ -229,21 +229,26 @@ func TestRenderTFVarsTmpl(t *testing.T) {
 			defer cancel()
 
 			gld := testutils.CheckFile(t, true, tc.goldenFile, "")
-			retry := 0
+			var cachedContents string
 			for {
 				re, err := r.Run(tmpl, w)
 				require.NoError(t, err)
 
 				if re.Complete {
 					// there may be a race with the consul services registering
 					// let's retry once.
-					if (string(gld) != string(re.Contents)) && retry < 1 {
-						retry++
+					contents := string(re.Contents)
+					if (string(gld) != contents) && cachedContents == "" {
+						cachedContents = contents
 						continue
 					}
 					assert.Equal(t, string(gld), string(re.Contents))
```

Comment:
Suggestion to reuse `contents` here instead of `string(re.Contents)`

Reply:
thank you! missed this